internal/security/redis/noauthconnect.go

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright 2020 Redis Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Andre Srinivasan
+ *******************************************************************************/
+
+package redis
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	dbContainer "github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/container"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+	dbInterfaces "github.com/edgexfoundry/edgex-go/internal/pkg/db/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db/redis"
+	"github.com/edgexfoundry/edgex-go/internal/security/redis/container"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+)
+
+func noAuthConnect(ctx context.Context, _ *sync.WaitGroup, startupTimer startup.Timer, dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+
+	config := container.ConfigurationFrom(dic.Get)
+
+	var dbClient dbInterfaces.DBClient
+	for startupTimer.HasNotElapsed() {
+		var err error
+		dbClient, err = redis.NewClient(db.Configuration{
+			Host: config.Databases["Primary"].Host,
+			Port: config.Databases["Primary"].Port,
+		}, lc)
+		if err == nil {
```

Comment:
What if error is that creds already set. Can you check for that and not doing any more processing??

Reply:
The service will now attempt both an unauthenticated and authenticated connection. Credentials are only set when the connection was unauthenticated.